### ISSUE TITLE
How to use batchsize for hypergraph

### ISSUE THREAD

**Issue by A1198482817a** [iMoonLab/DeepHypergraph#59]
Could you please provide guidance on how to utilize batch size  within the dhg.Hypergraph module?

**Comment by zjutnlper** [comment]
hello，do you find idea how to utilize batch size ？

**Comment by yifanfeng97** [comment]
DHG currently does not have built-in mini-batch support for hypergraph training (no equivalent of `dgl.batch`). The `Hypergraph` class operates on a single hypergraph at a time — the incidence matrix `H` is constructed for the entire graph.

**For now, the recommended approach is full-batch training** (the standard mode in all DHG examples):

```python
import dhg
import torch

# Load data
hg = dhg.Hypergraph(num_v, e_list)
X = torch.randn(num_v, feat_dim)

# Full-batch forward pass
model = dhg.models.HGNN(feat_dim, hid_dim, num_classes)
out = model(X, hg)
loss = criterion(out[train_mask], labels[train_mask])
```

If your graph is small enough to fit in GPU memory, full-batch is sufficient. For larger graphs, there are community workarounds:

1. **Subgraph sampling**: randomly sample sub-hypergraphs per step using vertex/hyperedge masks, build a smaller `Hypergraph` object each iteration.
2. **Manual batching**: concat multiple small hypergraphs feature matrices and use block-diagonal incidence matrices (similar to the DGL approach, but you handle the batching yourself).

Native mini-batch support is tracked in #34 and is a planned feature. This is also related to the graph-level classification batching question in #30.

**Comment by yifanfeng97** [comment]
Answered above. DHG currently supports full-batch training. Native mini-batch support is tracked in #34.
